pkg/sql/executor_ir.go

```diff
@@ -0,0 +1,424 @@
+	}
+
+	dropStmt := fmt.Sprintf("drop table if exists %s;", tableName)
+	if _, e := db.Exec(dropStmt); e != nil {
+		return fmt.Errorf("failed executing %s: %q", dropStmt, e)
+	}
+
+	fts, e := verify(predParsed, db)
+	if e != nil {
+		return e
+	}
+
+	var b bytes.Buffer
+	fmt.Fprintf(&b, "create table %s (", tableName)
+	for _, c := range predParsed.columns["feature_columns"] {
+		name, err := getExpressionFieldName(c)
+		if err != nil {
+			return err
+		}
+		typ, ok := fts.get(name)
+		if !ok {
+			return fmt.Errorf("createPredictionTable: Cannot find type of field %s", name)
+		}
+		stype, e := universalizeColumnType(db.driverName, typ)
+		if e != nil {
+			return e
+		}
+		fmt.Fprintf(&b, "%s %s, ", name, stype)
+	}
+
+	// TODO(Yancey1989): For the current implementation, the prediction result column
+	// type is derivated by the pred-select-statement, the better way is derivating
+	// the result column type by the prediction result.
+	typ, ok := fts.get(columnName)
+	if !ok {
+		// NOTE(typhoonzero): Clustering model may not have label in select statement, default use INT type
+		typ = "INT"
+	}
+	stype, e := universalizeColumnType(db.driverName, typ)
+	if e != nil {
+		return e
+	}
+	if db.driverName == "hive" {
+		fmt.Fprintf(&b, "%s %s) ROW FORMAT DELIMITED FIELDS TERMINATED BY \"\\001\" STORED AS TEXTFILE;", columnName, stype)
+	} else {
+		fmt.Fprintf(&b, "%s %s);", columnName, stype)
+	}
+
+	createStmt := b.String()
+	if _, e := db.Exec(createStmt); e != nil {
+		return fmt.Errorf("failed executing %s: %q", createStmt, e)
+	}
+	return nil
+}
+
+func loadModelMeta(pr *extendedSelect, db *DB, cwd, modelDir, modelName string) (*extendedSelect, fieldTypes, error) {
+	var m *model
+	var e error
+	if modelDir != "" {
+		m, e = loadTar(modelDir, cwd, modelName)
+	} else {
+		m, e = load(db, modelName, cwd)
+	}
+	if e != nil {
+		return nil, nil, fmt.Errorf("load %v", e)
+	}
+
+	// Parse the training SELECT statement used to train
+	// the model for the prediction.
+	tr, e := newParser().Parse(m.TrainSelect)
+	if e != nil {
+		return nil, nil, fmt.Errorf("parse: TrainSelect %v raise %v", m.TrainSelect, e)
+	}
+
+	if e := verifyColumnNameAndType(tr, pr, db); e != nil {
+		return nil, nil, fmt.Errorf("verifyColumnNameAndType: %v", e)
+	}
+
+	pr.trainClause = tr.trainClause
+	fts, e := verify(pr, db)
+	if e != nil {
+		return nil, nil, fmt.Errorf("verify: %v", e)
+	}
+
+	return pr, fts, nil
+}
+
+// ----------------------- useful for testing --------------------------
+
+func getDefaultSession() *pb.Session {
+	return &pb.Session{}
+}
+
+func errorPipe(err error) *PipeReader {
+	rd, wr := Pipe()
+	go func() {
+		defer wr.Close()
+		wr.Write(err)
+	}()
+	return rd
+}
+
+// RunSQLProgram run a raw SQL program (string list).
+func RunSQLProgram(sqlStatements []string, db *DB, modelDir string, session *pb.Session) *PipeReader {
+	connStr := fmt.Sprintf("%s://%s", db.driverName, db.dataSourceName)
+	cwd, err := ioutil.TempDir("/tmp", "sqlflow")
```

Comment:
I don't think we need to create a temporary directory for the whole SQL program.  Different SQL statements should have different working directories.